### PR TITLE
Feat run prettier on markdown

### PR DESCRIPTION
**Pull request by nathan-castlehow**
## Description
Adds support for Prettier in editor to "Prettify" Markdown.
Hotkey option available in config. Prettier Config json also available in config.
 Runs prettier over current markdown with configured prettier options.

prettier

## Issue fixed

#3039 

## Type of changes

- :white_circle: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
-  :radio_button: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

-  :radio_button: My code follows [the project code style](docs/code_style.md)
- :white_circle: I have written test for my code and it has been tested
- :radio_button: All existing tests have been passed
- :radio_button: I have attached a screenshot/video to visualize my change if possible

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#3039: [Feature Request] Format markdown notes with Prettier.](https://issuehunt.io/repos/53266139/issues/3039)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: browser/components/CodeEditor.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ import {generateInEditor, tocExistsInEditor} from 'browser/lib/markdown-toc-gene
 import markdownlint from 'markdownlint'
 import Jsonlint from 'jsonlint-mod'
 import { DEFAULT_CONFIG } from '../main/lib/ConfigManager'
+const prettier = require('prettier')
```
Comment (ZeroX-DG) [inline browser/components/CodeEditor.js]: Can you replace this with `import` statement?

File: browser/main/lib/ConfigManager.js [hunk 2]
```diff
@@ -65,7 +66,15 @@ export const DEFAULT_CONFIG = {
...
     enableSmartPaste: false,
     enableMarkdownLint: false,
-    customMarkdownLintConfig: DEFAULT_MARKDOWN_LINT_CONFIG
+    customMarkdownLintConfig: DEFAULT_MARKDOWN_LINT_CONFIG,
+    prettierConfig: ` {
+      "trailingComma": "es5",
+      "tabWidth": 4,
+      "semi": false,
+      "singleQuote": true,
+      "parser":"markdown"
```
Comment (ZeroX-DG) [inline browser/main/lib/ConfigManager.js]: Can you append this `parser` option at the end? Because this will never change so we should make it default.

Comment (nathan-castlehow) [reply browser/main/lib/ConfigManager.js]: @ZeroX-DG  do you mean as in don't have it in the options that show up in the config and then append the parser property before use?

File: browser/components/CodeEditor.js (hunk 1 above)
Comment (nathan-castlehow) [reply browser/components/CodeEditor.js]: Addressed in #bde357f

File: browser/main/lib/ConfigManager.js (hunk 2 above)
Comment (nathan-castlehow) [reply browser/main/lib/ConfigManager.js]: Assuming i understood the comment correctly this should be addressed in #1d59d89

Comment (ZeroX-DG) [reply browser/main/lib/ConfigManager.js]: Yep that's exactly what I wanted, thank you!

File: browser/components/CodeEditor.js (hunk 1 above)
Comment (ZeroX-DG) [reply browser/components/CodeEditor.js]: Great!

**Comment by ZeroX-DG** [comment]
One more small request, can you trigger the format table too? Currently it doesn't format the table.

**Comment by nathan-castlehow** [comment]
> One more small request, can you trigger the format table too? Currently it doesn't format the table.

Do you have sample markdown and expected results? I assume we are meaning its not correctly formatting a markdown table?

**Comment by ZeroX-DG** [comment]
I never used prettier to format markdown before so I didn't realized that they have support for table. However, in this case the table is not formatted with the default config. Here's my input:
```
|asd| asdasd|
|---|--------|
|asd|asdasdasd|
|dasd|ASdasdasd|
```

it should produce this:

```
| asd  | asdasd    |
| ---- | --------- |
| asd  | asdasdasd |
| dasd | ASdasdasd |
```

**Comment by nathan-castlehow** [comment]
@ZeroX-DG  i just tried the table you had and it works ok.
My default config is  
{
      "trailingComma": "es5",
      "tabWidth": 4,
      "semi": false,
      "singleQuote": true
  }

**Comment by nathan-castlehow** [comment]
@ZeroX-DG have you had a chance to look at this?

**Comment by ZeroX-DG** [comment]
@nathan-castlehow sorry for the late review. I found out about the reason behind this. If you have a line directly above the table then it will not format the table (or maybe format it wrongly).
```
asdasd
| asd  | asdasd|
| ---- | --------- |
| asd  | asdasdasd |
| dasd | ASdasdasd |
```

The code above will make the table to format wrongly. Is this a prettier bug or intended behavior?

**Comment by nathan-castlehow** [comment]
> @nathan-castlehow sorry for the late review. I found out about the reason behind this. If you have a line directly above the table then it will not format the table (or maybe format it wrongly).
> 
> ```
> asdasd
> | asd  | asdasd|
> | ---- | --------- |
> | asd  | asdasdasd |
> | dasd | ASdasdasd |
> ```
> 
> The code above will make the table to format wrongly. Is this a prettier bug or intended behavior?

@ZeroX-DG Hmm great question.
 When there is an empty line above the table (as below) it formats fine? Its either prettier intended behaviour or prettier bug i would say. Prettier uses remark-parse (https://github.com/remarkjs/remark/tree/master/packages/remark-parse) for dealing with the markdown.

 asdasd

| asd  | asdasd|
 | ---- | --------- |
 | asd  | asdasdasd |
 | dasd | ASdasdasd |

**Comment by ZeroX-DG** [comment]
Hmm...let's just say it's upstream issue for now. Can you fix the conflict so I can aprove it? @nathan-castlehow

**Comment by nathan-castlehow** [comment]
> Hmm...let's just say it's upstream issue for now. Can you fix the conflict so I can aprove it? @nathan-castlehow

@ZeroX-DG  Should be all sorted now. Still understanding the ropes of git (use tfs at work) so apologies in advance if there are any issues. Let me know if you need me to fix anything.

**Pull request merged**